Update Skia to 2025-02-05 (108cb2280ec2ce380d0b9421656fe413f1c08e81)

aperezdc opened the pull request:
#### 664c061aa42a7a78a67f66f722bc213eb9fdcf83
<pre>
Update Skia to 2025-02-05 (108cb2280ec2ce380d0b9421656fe413f1c08e81)
<a href="https://bugs.webkit.org/show_bug.cgi?id=287044">https://bugs.webkit.org/show_bug.cgi?id=287044</a>

Reviewed by Carlos Garcia Campos.

git log --oneline --pretty=&apos;%h %s&apos; 2cf673f5ebbb1c2a3688a73fa66c1d0110a7bd38..108cb2280ec2ce380d0b9421656fe413f1c08e81
108cb2280e Roll vulkan-deps from 6fe9d7e0521b to 9fa954563726 (6 revisions)
4bab884a7a [pdf] Do not scale miterLimit
a325d1e171 [graphite] list MtlGraphiteTypesPriv.h in bazel/gn sources
c9cb75cc19 [graphite] Add PrecompileContext::precompile option
9212eed0ce Merge 3 release notes into RELEASE_NOTES.md
d811590d05 Apply clang-tidy suggestion to GrProcessorUnitTest.cpp
1a2b02af18 [dawn][header] Use wgpu::StringView instead of const char*.
97741ae517 Update Skia milestone to 135
acbc3760a5 [dawn][headers] Use PassTimestampWrites.
7eea92a11f Roll vulkan-deps from 2e4b45a53a0e to 6fe9d7e0521b (10 revisions)
a6aa6c44d3 Roll ANGLE from 44b5f0a59f2a to 957bafb51066 (6 revisions)
ae41c09f89 Roll Skia Infra from f69f215a2ce4 to 16948351755e (9 revisions)
33823caf5b Roll Dawn from acd89d9f169a to 7132f397c4b2 (15 revisions)
d32ba2e5c5 [Graphite] Update Vulkan buffer barriers.
78147d092f Update skottie-base to include skottie-dev files
5a70a2623f [Ganesh] Update Intel GPU mappings in Vulkan backend
1c083af94e [graphite] Set up atlas-based clipping.
2e61d302c9 Roll vulkan-deps from de6d9c09f769 to 2e4b45a53a0e (30 revisions)
6c807bbebb [graphite] Fix ContextOptions.h #includes
b8bb51a5d9 Roll vulkan-deps from 938de304bdcb to de6d9c09f769 (14 revisions)
9a49cba9de [skottie] Drop motion blur support
d808779639 Make the Ganesh/Graphite DM precompilation options more distinct
690e06d65a [graphite] Add API to return Android-specific keys back to the user
629fcc5fbe Roll Dawn from c469d593acab to acd89d9f169a (1 revision)
d046f8abae Roll Skia Infra from 53594a79cc60 to f69f215a2ce4 (13 revisions)
421931e071 Manual roll ANGLE from 439dfe5e2c51 to 44b5f0a59f2a (14 revisions)
cc605d826d Manual roll Dawn from 6bc6697d10a4 to c469d593acab (10 revisions)
bef93b25d3 [Graphite] Use Dawn disable_lazy_clear_for_mapped_at_creation_buffer toggle
a3987e7d63 [skottie] Visible track matte adjustments
1d540887d4 Actually draw the new gradient interpolation types
9bd4453ec1 Skottie no longer checks version is set
6d3cae3feb Roll ANGLE from e330d9595d80 to 439dfe5e2c51 (8 revisions)
4bdf90faf7 Roll skcms from 6c5f4734f7da to 1e365691d01a (1 revision)
c1dc5033e7 Roll Skia Infra from 85d97f3cbf76 to 53594a79cc60 (5 revisions)
e0941791b8 Roll Dawn from 4dfae713222c to 6bc6697d10a4 (10 revisions)
ec8c632b8c [graphite] Use lock-free linked list return queue
568df9357f [Graphite] Add simple cache for reusing single Texture DescSets in Vk.
a9af2a74c5 [skottie] Handle missing SkSLEffect image inputs gracefully
284fd1df8b [skottie] Clamp path offset values
949cdb556b [graphite] Add serialization helpers for the Android-specific Precompile API
6fdad821ee [graphite] Use transfer color type in Mtl/DawnCaps
087e232890 Apply ClangTidy suggestions to ClipAtlasManager
e6daf687b5 [sksl] Move test from shared/ to errors/
3694c4878f Roll ANGLE from f231d94b280f to e330d9595d80 (11 revisions)
404ee727e3 Roll Skia Infra from fc6bd168cd2d to 85d97f3cbf76 (6 revisions)
074f23f821 Roll Dawn from 04c2ec3144a0 to 4dfae713222c (23 revisions)
008432ae30 Try to de-flake the TSAN_Graphite_Dawn_Metal job
a5ed95e3fb [Graphite] Fix Dawn NMOF compressed texture GM.
19819ab5f7 Roll skottie-base from 18c5fe2272c0 to 2f895d13711a
09f625266e Disable use of Dawn Validation on Release tool builds.
05830bf054 [graphite] Hook in atlas clip via PaintParams.
a0095479b3 [graphite] Call processReturnedResources() on occassion
68ff0e5ba0 Add remaining 4 color spaces as interpolation spaces
23705091c6 [graphite] Add ClipAtlasManager to AtlasProvider.
526c6ec833 Delete typeface after it has been registered to prevent memory leak
05f8ba7344 Roll vulkan-deps from 2c2860f3ab02 to 938de304bdcb (11 revisions)
371239eb12 Roll ANGLE from 0878d8543f62 to f231d94b280f (9 revisions)
0d9b2dbaec Roll Skia Infra from 9e2b0184e3cc to fc6bd168cd2d (5 revisions)
67619b60bd Roll Dawn from ba0f4a824014 to 04c2ec3144a0 (15 revisions)
dc44245ea9 [ganesh] Force gpu submit when there are too many render passes
fd9387e3a0 [dawn][headers] Removes default case for MapAsyncStatus.
c59d1f0a27 Remove references to decomissioned test devices
fc40278f12 SkCrabbyAvifCodec: Support repetition count
85e09e642c SkCrabbyAvifCodec: Clean up some TODOs
8a5ebe0b1e [graphite] Remove PremulAlphaColorFilter snippet
ca457248aa Split SkMaskFilter.cpp into two files
1c49281287 Try using less stack space for auto blitters
e314b4593c Use Apple&apos;s linear transformation equation for XR color types in SkRasterPipeline
f1e9a73c87 Roll vulkan-deps from ef9be82521e9 to 2c2860f3ab02 (2 revisions)
83dfedcc18 [graphite] Avoid int ops in universal color xform shader
43b8bae60b [graphite] Set up NonMSAAClip struct.
9006525891 [SkSL] Add check for unsized array in checkForOutOfRangeLiteral
e31d4adebc [graphite] Fix race condition in ThreadedPipelineCompilePurgingTest
ac975937ea Roll ANGLE from 8c3179402f7f to 0878d8543f62 (10 revisions)
dd163f1635 Roll Skia Infra from 417d9315f5b9 to 9e2b0184e3cc (7 revisions)
5427d44433 Roll Dawn from 539ed5c222e1 to ba0f4a824014 (20 revisions)
41ee18fc7d Roll SwiftShader from a87fa39f6446 to 86cf34f50cbe (1 revision)
6be067e436 [graphite] getImmutableSamplerInfo only takes in TextureInfo, not the whole proxy
d1a19e219e Roll vulkan-deps from 7425f0646395 to ef9be82521e9 (20 revisions)
f22419dbed Reland &quot;Use SkCrabbyAvifCodec for HEIC and AVIF on Android&quot;
e3878596cc SkCrabbyAvifCodec: Perform scaling after crop/subset
9451c53b58 Roll skcms from 12776964e26d to 6c5f4734f7da (1 revision)
777ca13dce Fixing the SkParagraph::getShapedLines method
b2df49bff8 Lower fuzz threshold in SkOpSegment even more
df23a06732 [Fontations] Synthesize x-height when not available from font
99994e16cd [graphite] Adjust assert in processReturnedResource()
84223bff71 Roll vulkan-deps from 4db7c3ecffea to 7425f0646395 (1 revision)
121ab59415 Roll ANGLE from d7ea4b2ce3c0 to 8c3179402f7f (2 revisions)
90a899cafa Roll Skia Infra from 3c369985ab80 to 417d9315f5b9 (7 revisions)
c4e70ced56 Roll Dawn from b9b4a37041de to 539ed5c222e1 (2 revisions)
d990cadd98 Roll vulkan-deps from a34400c4f6a0 to 4db7c3ecffea (1 revision)
42110712af [graphite] Fix Chrome roll/Resource.h header
408ac9e9be Manual roll Dawn from 363b4f1a0822 to b9b4a37041de (10 revisions)
3d20b817b2 Update test image to correct mistake
fec013f7b4 Roll vulkan-deps from 7625cc8362fc to a34400c4f6a0 (2 revisions)
0d91b1b40f Roll vulkan-deps from d06765b40a0b to 7625cc8362fc (4 revisions)
0fe231172a Manual roll ANGLE from 54b6968808c9 to d7ea4b2ce3c0 (7 revisions)
7456376d55 [graphite] Remove unref mutex and combine ref counts into single atomic
0f2833b305 Update SkColorType docs with bit patterns
a5c42ae18d Manual roll vulkan-deps from e54e6cd3a173 to d06765b40a0b (7 revisions)
ec32f0e67c SkCrabbyAvifCodec: Support rotation and mirroring
540db2bb3e [graphite] Fill out format tables
dc1a122178 Use templating to optimize std::function usage in Graphite PaintParams
3013d6d3f0 Roll skcms from 03700ab28aad to 12776964e26d (1 revision)
c5c98fc67c Roll vulkan-deps from fbbca36c20a6 to e54e6cd3a173 (6 revisions)
e6b5642def Roll ANGLE from c1214ec2c40d to 54b6968808c9 (5 revisions)
5891052b43 Roll Skia Infra from e0dc29520f99 to 3c369985ab80 (10 revisions)
29ff0ff0b4 Roll SwiftShader from 73053a0586b0 to a87fa39f6446 (4 revisions)
1c2a24f65d Roll Dawn from 138d7d354e31 to 363b4f1a0822 (2 revisions)
e50edb2cdb Manual roll ANGLE from 40267ddca2b8 to c1214ec2c40d (8 revisions)
06849eb99a Manual roll vulkan-deps from 5eba65d8d005 to fbbca36c20a6 (8 revisions)
8b035c3a79 [graphite] Consolidate determinations of dst texture read strategies
c8aba6a056 [graphite] Add shader snippet for ClipAtlas.
bb03124471 Skip empty chunks in SkPngCompositeChunkReader
57c433af1a Manual roll Dawn from d60d8683c9a5 to 138d7d354e31 (10 revisions)
dcf659997c [graphite] Specialization for sRGB color xforms
58ccd4f768 [Graphite] Add SkSurface::makeTemporaryImage API.
5061262298 Roll vulkan-deps from b15753b47ae5 to 5eba65d8d005 (4 revisions)
7cd26c8c4d [dawn][headers] Removes explicit MapAsync::Unknown.
3b431953b9 Add new gn location to .gitignore
b34b6f115f [Graphite] Stop letting label back to Scratch in ResourceCache.
eb9ff8e427 Use 50% load factor for `fDigestForPackedGlyphID` table
9d51bc2e06 Roll ANGLE from 3734b432018d to 40267ddca2b8 (15 revisions)
a3523a7b60 Roll Skia Infra from 11f410b4c438 to e0dc29520f99 (4 revisions)
b808498a80 Roll Dawn from 93bf2eaa5aa6 to d60d8683c9a5 (16 revisions)
ba77260fbc Roll vulkan-deps from 64530d41c091 to b15753b47ae5 (6 revisions)
c3b763fb28 Revert &quot;Use SkCrabbyAvifCodec for HEIC and AVIF on Android&quot;
d2db696754 [bzl] Fix //dm build
76ee21c63e [graphite] Fix up RasterPathUtils for clipping.
4ff2e3a8fe [rust png] Parse orientation from the `eXIf` chunk.
73a5aec9ad Add some documentation to the mesh_updates GM
6532b64a9e Update Skia SkCanvas docs to correct SkSurfaces methods
6f40e3c2aa Manual roll Dawn from fe1a3b8c65ed to 93bf2eaa5aa6 (8 revisions)
953cc77ad1 Manual roll SwiftShader from b99f866d5a5c to 73053a0586b0 (1 revision)
46407f56ff [infra] Add jobs for new MacMinis
c815a9b265 Manual Dawn roll
7a3eedf917 Fetch gn to third_party/gn
b52b705646 Add missing include in SkDrawTypes
03ff690028 [Fontations-backend] Roll Fontations, Skrifa to 0.26.5
8b20be96b4 Roll vulkan-deps from cbf4ac2af7a5 to 64530d41c091 (7 revisions)
96bbf45d79 Roll ANGLE from c7cfbbc90295 to 3734b432018d (7 revisions)
836fdfc670 Roll Skia Infra from 04b04dcc4949 to 11f410b4c438 (8 revisions)
22640ab9e3 Introduce SkDrawCoverage instead of error-prone boolean
7c76a89593 Mark two SkDevice subclasses as final
46ecd1d904 [graphite] Rename Resource timestamp to last use token
8a0d69cd06 [graphite] Remove Mac10.15.1 Graphite_Dawn_Metal jobs
c5dd82ebdb Roll vulkan-deps from 8d5a282ee9ba to cbf4ac2af7a5 (2 revisions)
fdfd5e5174 [graphite] Organize members of ResourceCache
5a6323775b [Fontations] Protect path retrieval arrays with lock
464a376e27 Roll vulkan-deps from a9e496961117 to 8d5a282ee9ba (6 revisions)
40c262ff88 [Fontations] Synthethize cap height
3050c2ce16 [graphite] Add static IDs for RenderSteps
e13090e31f Roll ANGLE from 0dabfe5a6ac5 to c7cfbbc90295 (9 revisions)
06677fe6a7 Roll Skia Infra from e0480d491b7b to 04b04dcc4949 (2 revisions)
60441f8c43 Roll Dawn from 50824d5923fa to c346956b0ef9 (8 revisions)
57adbff41b Roll vulkan-deps from 55e70e1178e9 to a9e496961117 (11 revisions)
7f19d763e0 Roll vulkan-deps from 127cb2fa829b to 55e70e1178e9 (1 revision)
b9b67f9b36 Roll ANGLE from 29731140b15f to 0dabfe5a6ac5 (25 revisions)
dac808134f Roll Skia Infra from 28df2c051126 to e0480d491b7b (8 revisions)
d0ebb49951 Roll Dawn from 0c8f10626e6c to 50824d5923fa (3 revisions)
ffd7fec8b6 [bzl] Fix G3 rolls
dca0ac967c Manual roll SwiftShader from f6ad529e86e4 to b99f866d5a5c (2 revisions)
cbca313620 Manual roll Dawn from 9185a4f4c58b to 0c8f10626e6c (15 revisions)
a9e8d004f0 Roll vulkan-deps from f840db3658c0 to 127cb2fa829b (14 revisions)
73341093dd Register FreeType typefaces in debugger
57abed1ab7 [bzl] Enable layering_check for Mac frameworks
5d55822e00 [graphite] Organize members in Resource.h
377ace7260 [bzl] Fix Viewer builds on Mac
d4b56ba603 Apply misc cleanups to mask filter and resource cache
c2b103c89e Fix and annotate octopus generator fiddles
eb680e6843 Roll vulkan-deps from 7e22bcde70ab to f840db3658c0 (3 revisions)
05dfc6a47c Roll Skia Infra from 0f74c98b09c6 to 28df2c051126 (6 revisions)
af80f4bb33 Roll Dawn from 617d042d4bb2 to 9185a4f4c58b (11 revisions)
1c0c5e9016 [bzl] Bump arm64 Mac toolchain to 17.0.6
d03e18f549 Roll vulkan-deps from c4ff2df7f2af to 7e22bcde70ab (4 revisions)
ea5d3fe55b [graphite] Use true scratch resources in ScratchResourceManager
367c280216 Manual roll ANGLE from de5286fa682b to 29731140b15f (2 revisions)
0a0fb30708 Deprecate (and reimpl) TightBounds
40108bcf85 Roll vulkan-deps from bd8887349203 to c4ff2df7f2af (1 revision)
05a21cbb87 [skottie] Support arbitrary displacement map sources
120379fefa Roll Skia Infra from 982a528a4385 to 0f74c98b09c6 (1 revision)
7dcc6fd6a6 Roll vulkan-deps from 3f678cea6621 to bd8887349203 (11 revisions)
d51aa116b3 Roll Skia Infra from 2f565c11a12c to 982a528a4385 (2 revisions)
efa5810875 Roll Skia Infra from 0529441d608f to 2f565c11a12c (8 revisions)
3978daa523 Roll Dawn from b7143cc4f3bf to 617d042d4bb2 (17 revisions)
488804de2c [graphite] Specialization for un/premul color xforms
2d01763761 [graphite] Add caching functionality to ClipAtlasManager
871e99c792 SkCrabbyAvifCodec: Support subsets
fb6df6a5fb Stop int overflow in TiledTextureUtils.cpp
19e7b61e39 Manual roll ANGLE from 0a207b65f60f to de5286fa682b (20 revisions)
06eb91403a [ganesh] Remove deprecated AutoCheckFlush interface.
638c27f067 [graphite] Consolidate path atlas translations into a single place.
c23e5c61e5 Roll vulkan-deps from bc80d88f6102 to 3f678cea6621 (1 revision)
8d366b3107 Break up skparagraph GNI lists
4076192f74 Extend the lifetime of SkPngCompositeChunkReader
1769d784c6 Use SkCrabbyAvifCodec for HEIC and AVIF on Android
78b812cc74 [graphite] Add Scratch shareable concept to ResourceCache
780e06a39d [bazel] Use xcrun to locate the macOS SDK
a6e93c6fc2 Roll Skia Infra from 9e9b5c63a6f8 to 0529441d608f (6 revisions)
e3dd30787e Roll Dawn from bfcb830e04ff to b7143cc4f3bf (16 revisions)
f71e1ff094 Roll vulkan-deps from d47ae2be2781 to bc80d88f6102 (10 revisions)
839e579f26 Roll skottie-base from 659bf938549f to 18c5fe2272c0
427dc976d0 Add more suppressions for ThreadedPipelineCompilePurgingTest
8d10158ffd Give RecreateSKPs 8h
123e9e7550 Roll SwiftShader from e76961fac58c to f6ad529e86e4 (4 revisions)
8b8ff96b13 Update CMake
e7b8d07885 Roll vulkan-deps from 5f7e87250bc4 to d47ae2be2781 (2 revisions)
b0427017da Roll Skia Infra from 90fa48bf59cb to 9e9b5c63a6f8 (5 revisions)
754b2cf745 Roll Dawn from 817f471a8e17 to bfcb830e04ff (26 revisions)
5b904e1b13 Roll vulkan-deps from a0c6bbad67c9 to 5f7e87250bc4 (5 revisions)
63a37d183a [rust png] Encode `pixmap.colorSpace()` as `iCCP` chunk.
c24b41eebc [graphite] Suppress threaded Precompile tests on Dawn/GLES and Dawn/Metal
3ff3711046 [bmp] Fix swizzler initialization
c6700809d2 [graphite] PrimitiveColor composes colorspace transform. Support skipColorXform for drawVertices
22f9ea65f4 [graphite] Add Pipeline cache purging
539c311a3a Roll ANGLE from f64fef257ef0 to 0a207b65f60f (14 revisions)
3b0746ea21 Roll vulkan-deps from 8540477b5d95 to a0c6bbad67c9 (1 revision)
528aa95879 Roll Skia Infra from c67b4fc704ca to 90fa48bf59cb (3 revisions)
6ddc7ba220 Roll Dawn from 8c8e27355e12 to 817f471a8e17 (13 revisions)
3531a78f28 Roll vulkan-deps from 199e36d15635 to 8540477b5d95 (16 revisions)
1398cbd6b7 Roll vulkan-deps from e264c58f9b0f to 199e36d15635 (3 revisions)
dba7f46122 Automatically scale animated webp post-decode in Android
5f2e30e0c8 Render fonts right that require hinting, as detected by Skrifa
95ed528a21 Roll ANGLE from 71980a30e08c to f64fef257ef0 (9 revisions)
66291bd9b5 Roll Skia Infra from 3c89288303c9 to c67b4fc704ca (6 revisions)
fec7d8df29 Roll Dawn from ced2fcb5d664 to 8c8e27355e12 (12 revisions)
b660e30f0a Roll vulkan-deps from 1fa2eb257bf0 to e264c58f9b0f (4 revisions)
cf47a26b74 Combine SkColorShader and SkColor4Shader
04a6578d2b [graphite] Track whether or not the Resource is reusable on the Resource
9658353bf2 Loosen G3 unsupported API Presubmit check
1d532cb013 Allow more slop in TypefaceMacVariation test
8a79cd890f SkCrabbyAvifCodec: Return correct image type
05743a92fd Revert &quot;[ganesh] Unit test for flush of color writing for advanced blend modes&quot;
9b556469c2 Revert &quot;Update suppression for DMSAA_blendmodes tests&quot;
65b12d1c82 Update suppression for DMSAA_blendmodes tests
04f0c6ef20 Add presubmit rule to check for unuspported G3 APIs
36feb25c12 [ganesh] Unit test for flush of color writing for advanced blend modes
f43d161439 [Fontations-backend] Roll Fontations, Skrifa to 0.26.3
6223c052f3 Roll vulkan-deps from 254f6951b8dc to 1fa2eb257bf0 (4 revisions)
083179bb80 Roll ANGLE from a51899d8733d to 71980a30e08c (6 revisions)
623fda5ab1 Roll Skia Infra from 2a7b1c4b1bcd to 3c89288303c9 (6 revisions)
918ba2965a Roll Dawn from 2d7fbc4b1f1b to ced2fcb5d664 (9 revisions)
c1886c4e47 Reland &quot;[rust png] Report null color profile if only &quot;neutral&quot; gamma is encoded.&quot;
b44aa9b66f [graphite] Move Shareable state off key over to Resource
3ac73808d1 [ganesh] Fix the stack depth counter in GrTriangulator.
66fe3accc7 Reland &quot;[graphite] Consolidate ycbcr sampler handling in Dawn backend&quot;
b95665caaf Revert &quot;Replace SkSharedMutex with std::shared_mutex&quot;
e17eb688f4 Roll vulkan-deps from 5f05f0592633 to 254f6951b8dc (1 revision)
42a99357fb [Fontations-backend] Roll Fontations, Skrifa to 0.26.2
f8ba688521 Reduce attempts in SkOpSegment::spansNearby
40d438d766 Roll ANGLE from 465299fba5da to a51899d8733d (12 revisions)
bf74acc0a1 Roll Skia Infra from e0f9392b4eda to 2a7b1c4b1bcd (7 revisions)
fc8d77f726 Roll Dawn from 3183110b30f8 to 2d7fbc4b1f1b (10 revisions)
8bbd583477 Roll vulkan-deps from a652576bff1b to 5f05f0592633 (12 revisions)
c09407add3 Make defensive copies of SkPath less frequently when drawing them
e944b50469 Revert &quot;[rust png] Report null color profile if only &quot;neutral&quot; gamma is encoded.&quot;
8aed02367a [skottie] Support nonsequential track mattes
7cf08825d7 Revert &quot;[graphite] Consolidate ycbcr sampler handling in Dawn backend&quot;
0d70ea7bf0 [rust png] Report null color profile if only &quot;neutral&quot; gamma is encoded.
e995e9f190 [graphite] ResourceProvider defines Sampler resource key
3d1ecf63e5 Reland &quot;Limit stack calling depth from GrTriangulator::mergeCollinearEdges().&quot;
42e38a8610 [graphite] Consolidate ycbcr sampler handling in Dawn backend
8d5fabf4f8 [graphite] Consolidate vulkan ycbcr conversion handling
9b3395edd0 Roll vulkan-deps from e2e618fdd93a to a652576bff1b (8 revisions)
52f7ba884f Merge 9 release notes into RELEASE_NOTES.md
50e6fe5901 Update Skia milestone to 134
e71ab89812 Roll ANGLE from d65751b40f7e to 465299fba5da (13 revisions)
f820e6f330 Roll Skia Infra from 783200f5fd9e to e0f9392b4eda (3 revisions)
c99770530d Roll Dawn from 3f908f3907e5 to 3183110b30f8 (10 revisions)
5310fc642c Roll vulkan-deps from 0cf38d98a8ba to e2e618fdd93a (7 revisions)
9799dcab37 [webp] Use the same RP blender for all lines
279fde045c GraphiteDawn: Don&apos;t enable SSBO in Compat Mode.
77f4941092 Set underlying type of SkFontStyle::Slant
4a7628f0c8 In play_skottie.cpp example check that surface was allocated
84d9fd6e3b SkFontMgr_FCI refactoring
20164f3db7 [Graphite] Implement Vulkan query to know lazy allocation size.
3ee39f3584 Allow MSKP benches to throttle frames on GPU.
472faa80de Roll ANGLE from b225419fd227 to d65751b40f7e (2 revisions)
1c76d8b280 Roll vulkan-deps from 85de9aa78bc8 to 0cf38d98a8ba (2 revisions)
58c6de0423 Roll Skia Infra from 6fd1ead44af5 to 783200f5fd9e (3 revisions)
71e698aedf Roll Dawn from 28858cfac0b9 to 3f908f3907e5 (1 revision)
cb18e028c4 Roll vulkan-deps from d2e10146ec6f to 85de9aa78bc8 (1 revision)
b4613bade5 Roll vulkan-deps from a09312dc04d3 to d2e10146ec6f (1 revision)
babfedf497 Manual roll Dawn from 7f24ec953f26 to 28858cfac0b9 (7 revisions)
975788ea90 [graphite] Fix clients that depend on MtlGraphiteTypes.h
34c784a292 Remove approximateFilteredBounds from SkMaskFilter
5857fc2fe4 Use skia_rules.bzl instead of macros.bzl where possible
ac9c72dcf0 [graphite] Metal backend type utility functions callable from C++
fbc37088ac Ganesh uses default bazel obj copts, Graphite explicit about no ARC.
df3f49098f Expose skottie for external Bazel clients.
b8e83110f0 Roll vulkan-deps from 57295eb4b939 to a09312dc04d3 (1 revision)
44ef31b19d Roll ANGLE from 9d9e3efce077 to b225419fd227 (2 revisions)
fffbc97d9d Roll Skia Infra from ec5e5e9d35c8 to 6fd1ead44af5 (1 revision)
efbf755499 Roll Dawn from 5b33b8d8385d to 7f24ec953f26 (4 revisions)
8cfeb42c62 Roll vulkan-deps from b7e7b5c365d1 to 57295eb4b939 (3 revisions)
3123598acb [pdf] Allow simple /StructElem /K
24365abf85 [rust png] Performance: Cache and reuse `fColorTable` if possible.
d2e8363d13 [rust png] Avoid doing alpha premultiplication twice.
fe8bac8f22 Remove unneeded resource refs on Vulkan CommandBuffer.
8eebb997a5 Update viewer builds to work on newest android devices.
65d7bff9d1 [rust png] Use `u32` =&gt; `f32` conversion that matches Blink.
68d4407438 Roll vulkan-deps from f45ce3e3fa15 to b7e7b5c365d1 (5 revisions)
8670a9edba Roll ANGLE from 6015e7305d15 to 9d9e3efce077 (3 revisions)
095e1a7961 Roll Skia Infra from d359a715b7ed to ec5e5e9d35c8 (1 revision)
bb06d0aa57 Roll Dawn from 28b186286761 to 5b33b8d8385d (2 revisions)
693030a1df Roll vulkan-deps from 04a50a3dd7b9 to f45ce3e3fa15 (1 revision)
e99b226c3d Roll ANGLE from 39882b2b735f to 6015e7305d15 (2 revisions)
31d19df840 Roll Skia Infra from d10e2fbb052e to d359a715b7ed (2 revisions)
d59c150674 Roll Dawn from e5701af2e856 to 28b186286761 (1 revision)
e045d7ef5e Roll vulkan-deps from bfa23916c97e to 04a50a3dd7b9 (2 revisions)
09f8ab0c9b Roll vulkan-deps from 93cfcbf427b3 to bfa23916c97e (2 revisions)
5e8c3cb793 Roll ANGLE from 017a9382befe to 39882b2b735f (1 revision)
89f1df000d Roll Skia Infra from 9dab19606d40 to d10e2fbb052e (4 revisions)
d1c208fb5e Roll Dawn from dc4935b883dd to e5701af2e856 (2 revisions)
c260865988 Manual roll ANGLE from c0ee7b203bad to 017a9382befe (40 revisions)
efe8bac7d1 Roll vulkan-deps from 4cc576cd0e3d to 93cfcbf427b3 (1 revision)
787f004498 Roll Dawn from db55f5957411 to dc4935b883dd (5 revisions)
fddefe73c5 Roll Skia Infra from 1ed863c4385b to 9dab19606d40 (6 revisions)
0ec5145fca Roll SwiftShader from 118f577b4188 to e76961fac58c (1 revision)
b5839e6326 Roll vulkan-deps from 6455a044bee9 to 4cc576cd0e3d (1 revision)
e107cb001d Roll vulkan-deps from 2b029290ba80 to 6455a044bee9 (4 revisions)
33a9a4ffae Roll vulkan-deps from b7874f7020d2 to 2b029290ba80 (1 revision)
44b9707fc1 Roll vulkan-deps from 2ceb995d091f to b7874f7020d2 (1 revision)
7a4ef0b9a2 Roll Skia Infra from b95a7b5a23b2 to 1ed863c4385b (5 revisions)
db37ded7ce Roll Dawn from 76129f64155f to db55f5957411 (2 revisions)
22127449d7 Roll Skia Infra from 56a20669db21 to b95a7b5a23b2 (1 revision)
9ce064f60d Roll Dawn from 45926030e9e1 to 76129f64155f (2 revisions)
ccefbb7a59 Roll Skia Infra from b0b99f248aab to 56a20669db21 (1 revision)
daecd808dc WebGPU: Use VertexStepMode Undefined instead of VertexBufferNotUsed
7b16ec6d9c Roll Skia Infra from 2ad74dabbee5 to b0b99f248aab (6 revisions)
c0c3d52f29 Roll Dawn from 23bb5c065341 to 45926030e9e1 (3 revisions)
bdbc037fd6 Roll vulkan-deps from d9e522973bf6 to 2ceb995d091f (2 revisions)
1c9c95eb6f Keep last resort glyphs and positions together
e0114876f8 Only output last valid scan of progressive JPEG decode
79f1fd9a9d [infra] Add jobs for new MacBookPro15,3 machines
0440559439 Roll vulkan-deps from dd42fc42309c to d9e522973bf6 (3 revisions)
46df6fed41 Roll Skia Infra from 0abec67a371b to 2ad74dabbee5 (7 revisions)
f0bf1fd96d Roll Dawn from ef00af577667 to 23bb5c065341 (1 revision)
1949fee840 Roll vulkan-deps from 0f01167cd48b to dd42fc42309c (1 revision)
d5f0cf5a7c Roll vulkan-deps from f6642af20ac1 to 0f01167cd48b (2 revisions)
cf67bf31e0 Manual roll Dawn from bd302da5fe83 to ef00af577667 (10 revisions)
47957be04e Revert &quot;Limit stack calling depth from GrTriangulator::mergeCollinearEdges().&quot;
eb33dd41db Roll vulkan-deps from b51f6c0aa71b to f6642af20ac1 (11 revisions)
a38c4ae028 Limit stack calling depth from GrTriangulator::mergeCollinearEdges().
005f65a914 [graphite] Disable use of CommandBufferTimestamps for dawn-mac
40959c4dbc [Graphite] Add support in the ResourceCache for change a Resources size.
a0c37ec4ca [graphite] Adjust raster path key to include clip origin.
a15d02f775 Update the test &apos;FontMgr_Iter&apos;
99dd91b4fe Remove CoreText 14 and 15 work arounds
2521ff091d [infra] Move to Android devices in the new lab
70d482d240 Roll vulkan-deps from 33f530ef3285 to b51f6c0aa71b (3 revisions)
c88fc17d1d Roll Dawn from c3cd3b4a6d40 to bd302da5fe83 (7 revisions)
38461b444d Roll Skia Infra from fb8dd36529f6 to 0abec67a371b (9 revisions)
e35527e40d Roll vulkan-deps from e616c9b1ad31 to 33f530ef3285 (4 revisions)
efebb9d95e Make size pre-allocated for SkCanvas::save() controllable in SkUserConfig
96b37e2a3e [viewer] Check for valid WindowContext in Window::supportsGpuTimer
8f33adb1ac Adjust slant value for &apos;CTFontDescriptor&apos; to improve font matching
ee9f7aec12 fiddle_main.h to not include skia.h
d746d5faaa Add SK_NO_SANITIZE_CFI that has effect when building with Clang
8c518c6d3a [graphite][Dawn] Add missing attribute from YCbCrVkDescriptor comparison
60f1c30d07 Clarify size of SkCanvas&apos;s stack allocation
69ebfded97 [infra] Replace RadeonVega6 jobs with RadeonVega8
e5296cebaa [graphite] Add Vulkan push constant support
1d9cace0d9 Manual roll vulkan-deps from 491933b72060 to e616c9b1ad31 (7 revisions)
e3b2835099 Roll vulkan-deps from ae8417f428a1 to 491933b72060 (5 revisions)
20e3d71981 Roll Skia Infra from 7544f7c1ccbb to fb8dd36529f6 (10 revisions)
44e33b331d Roll Dawn from cc2acb18f2a8 to c3cd3b4a6d40 (13 revisions)
10a65a187b Roll SwiftShader from 23b93c7cfe27 to 118f577b4188 (1 revision)
b2f382eaae [rust png] More granular gni list: `skia_encode_libpng_srcs`.
cdd807647d [rust png] Update `cxx` from 1.0.131 to 1.0.135.
85baab670f Remove skiaserve and libmicrohttpd
a6ea45cb42 [graphite] Establish exclusive descriptor set for Vulkan input attachments
248b5242d0 Support correct flushing for Graphite in MSKPBench
4a9d5e24dd Roll skottie-base from 32e4afed6d80 to 659bf938549f
fc205a7656 [viewer] Fix captured reference to stack variable
8f2a29de51 Fix TODO in SkRRect about dense packing
492a9e761c Roll vulkan-deps from dbec774c7f68 to ae8417f428a1 (2 revisions)

Canonical link: <a href="https://commits.webkit.org/289878@main">https://commits.webkit.org/289878@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/09bea89acacc54e87d03eb3a365993b5f2e9c19a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/88249 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/7766 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/42678 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/93203 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/38999 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/90300 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/8150 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/15947 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/68081 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/25797 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/91251 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/6223 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/79824 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/48449 "Found 1 new API test failure: /TestJSC:/jsc/options (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/5994 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/34233 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/38107 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/81047 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/76377 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/35118 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/95048 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/87025 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/15419 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/11299 "Found 3 new test failures: imported/w3c/web-platform-tests/html/capability-delegation/delegate-fullscreen-request-subframe-same-origin.https.tentative.html imported/w3c/web-platform-tests/screen-orientation/lock-basic.html media/modern-media-controls/tracks-support/audio-multiple-tracks.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/76944 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/15675 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/75681 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/76190 "run-api-tests-without-change (failure)") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/20573 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/19000 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/8446 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/13784 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/15437 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/109518 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/15179 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/26337 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/18625 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/16961 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

GeorgesStavracas reviewed (approved):
Not a reviewer but LGTM nonetheless

aperezdc committed:
Update Skia to 2025-02-05 (108cb2280ec2ce380d0b9421656fe413f1c08e81)
https://bugs.webkit.org/show_bug.cgi?id=287044

Reviewed by Carlos Garcia Campos.

git log --oneline --pretty='%h %s' 2cf673f5ebbb1c2a3688a73fa66c1d0110a7bd38..108cb2280ec2ce380d0b9421656fe413f1c08e81
108cb2280e Roll vulkan-deps from 6fe9d7e0521b to 9fa954563726 (6 revisions)
4bab884a7a [pdf] Do not scale miterLimit
a325d1e171 [graphite] list MtlGraphiteTypesPriv.h in bazel/gn sources
c9cb75cc19 [graphite] Add PrecompileContext::precompile option
9212eed0ce Merge 3 release notes into RELEASE_NOTES.md
d811590d05 Apply clang-tidy suggestion to GrProcessorUnitTest.cpp
1a2b02af18 [dawn][header] Use wgpu::StringView instead of const char*.
97741ae517 Update Skia milestone to 135
acbc3760a5 [dawn][headers] Use PassTimestampWrites.
7eea92a11f Roll vulkan-deps from 2e4b45a53a0e to 6fe9d7e0521b (10 revisions)
a6aa6c44d3 Roll ANGLE from 44b5f0a59f2a to 957bafb51066 (6 revisions)
ae41c09f89 Roll Skia Infra from f69f215a2ce4 to 16948351755e (9 revisions)
33823caf5b Roll Dawn from acd89d9f169a to 7132f397c4b2 (15 revisions)
d32ba2e5c5 [Graphite] Update Vulkan buffer barriers.
78147d092f Update skottie-base to include skottie-dev files
5a70a2623f [Ganesh] Update Intel GPU mappings in Vulkan backend
1c083af94e [graphite] Set up atlas-based clipping.
2e61d302c9 Roll vulkan-deps from de6d9c09f769 to 2e4b45a53a0e (30 revisions)
6c807bbebb [graphite] Fix ContextOptions.h #includes
b8bb51a5d9 Roll vulkan-deps from 938de304bdcb to de6d9c09f769 (14 revisions)
9a49cba9de [skottie] Drop motion blur support
d808779639 Make the Ganesh/Graphite DM precompilation options more distinct
690e06d65a [graphite] Add API to return Android-specific keys back to the user
629fcc5fbe Roll Dawn from c469d593acab to acd89d9f169a (1 revision)
d046f8abae Roll Skia Infra from 53594a79cc60 to f69f215a2ce4 (13 revisions)
421931e071 Manual roll ANGLE from 439dfe5e2c51 to 44b5f0a59f2a (14 revisions)
cc605d826d Manual roll Dawn from 6bc6697d10a4 to c469d593acab (10 revisions)
bef93b25d3 [Graphite] Use Dawn disable_lazy_clear_for_mapped_at_creation_buffer toggle
a3987e7d63 [skottie] Visible track matte adjustments
1d540887d4 Actually draw the new gradient interpolation types
9bd4453ec1 Skottie no longer checks version is set
6d3cae3feb Roll ANGLE from e330d9595d80 to 439dfe5e2c51 (8 revisions)
4bdf90faf7 Roll skcms from 6c5f4734f7da to 1e365691d01a (1 revision)
c1dc5033e7 Roll Skia Infra from 85d97f3cbf76 to 53594a79cc60 (5 revisions)
e0941791b8 Roll Dawn from 4dfae713222c to 6bc6697d10a4 (10 revisions)
ec8c632b8c [graphite] Use lock-free linked list return queue
568df9357f [Graphite] Add simple cache for reusing single Texture DescSets in Vk.
a9af2a74c5 [skottie] Handle missing SkSLEffect image inputs gracefully
284fd1df8b [skottie] Clamp path offset values
949cdb556b [graphite] Add serialization helpers for the Android-specific Precompile API
6fdad821ee [graphite] Use transfer color type in Mtl/DawnCaps
087e232890 Apply ClangTidy suggestions to ClipAtlasManager
e6daf687b5 [sksl] Move test from shared/ to errors/
3694c4878f Roll ANGLE from f231d94b280f to e330d9595d80 (11 revisions)
404ee727e3 Roll Skia Infra from fc6bd168cd2d to 85d97f3cbf76 (6 revisions)
074f23f821 Roll Dawn from 04c2ec3144a0 to 4dfae713222c (23 revisions)
008432ae30 Try to de-flake the TSAN_Graphite_Dawn_Metal job
a5ed95e3fb [Graphite] Fix Dawn NMOF compressed texture GM.
19819ab5f7 Roll skottie-base from 18c5fe2272c0 to 2f895d13711a
09f625266e Disable use of Dawn Validation on Release tool builds.
05830bf054 [graphite] Hook in atlas clip via PaintParams.
a0095479b3 [graphite] Call processReturnedResources() on occassion
68ff0e5ba0 Add remaining 4 color spaces as interpolation spaces
23705091c6 [graphite] Add ClipAtlasManager to AtlasProvider.
526c6ec833 Delete typeface after it has been registered to prevent memory leak
05f8ba7344 Roll vulkan-deps from 2c2860f3ab02 to 938de304bdcb (11 revisions)
371239eb12 Roll ANGLE from 0878d8543f62 to f231d94b280f (9 revisions)
0d9b2dbaec Roll Skia Infra from 9e2b0184e3cc to fc6bd168cd2d (5 revisions)
67619b60bd Roll Dawn from ba0f4a824014 to 04c2ec3144a0 (15 revisions)
dc44245ea9 [ganesh] Force gpu submit when there are too many render passes
fd9387e3a0 [dawn][headers] Removes default case for MapAsyncStatus.
c59d1f0a27 Remove references to decomissioned test devices
fc40278f12 SkCrabbyAvifCodec: Support repetition count
85e09e642c SkCrabbyAvifCodec: Clean up some TODOs
8a5ebe0b1e [graphite] Remove PremulAlphaColorFilter snippet
ca457248aa Split SkMaskFilter.cpp into two files
1c49281287 Try using less stack space for auto blitters
e314b4593c Use Apple's linear transformation equation for XR color types in SkRasterPipeline
f1e9a73c87 Roll vulkan-deps from ef9be82521e9 to 2c2860f3ab02 (2 revisions)
83dfedcc18 [graphite] Avoid int ops in universal color xform shader
43b8bae60b [graphite] Set up NonMSAAClip struct.
9006525891 [SkSL] Add check for unsized array in checkForOutOfRangeLiteral
e31d4adebc [graphite] Fix race condition in ThreadedPipelineCompilePurgingTest
ac975937ea Roll ANGLE from 8c3179402f7f to 0878d8543f62 (10 revisions)
dd163f1635 Roll Skia Infra from 417d9315f5b9 to 9e2b0184e3cc (7 revisions)
5427d44433 Roll Dawn from 539ed5c222e1 to ba0f4a824014 (20 revisions)
41ee18fc7d Roll SwiftShader from a87fa39f6446 to 86cf34f50cbe (1 revision)
6be067e436 [graphite] getImmutableSamplerInfo only takes in TextureInfo, not the whole proxy
d1a19e219e Roll vulkan-deps from 7425f0646395 to ef9be82521e9 (20 revisions)
f22419dbed Reland "Use SkCrabbyAvifCodec for HEIC and AVIF on Android"
e3878596cc SkCrabbyAvifCodec: Perform scaling after crop/subset
9451c53b58 Roll skcms from 12776964e26d to 6c5f4734f7da (1 revision)
777ca13dce Fixing the SkParagraph::getShapedLines method
b2df49bff8 Lower fuzz threshold in SkOpSegment even more
df23a06732 [Fontations] Synthesize x-height when not available from font
99994e16cd [graphite] Adjust assert in processReturnedResource()
84223bff71 Roll vulkan-deps from 4db7c3ecffea to 7425f0646395 (1 revision)
121ab59415 Roll ANGLE from d7ea4b2ce3c0 to 8c3179402f7f (2 revisions)
90a899cafa Roll Skia Infra from 3c369985ab80 to 417d9315f5b9 (7 revisions)
c4e70ced56 Roll Dawn from b9b4a37041de to 539ed5c222e1 (2 revisions)
d990cadd98 Roll vulkan-deps from a34400c4f6a0 to 4db7c3ecffea (1 revision)
42110712af [graphite] Fix Chrome roll/Resource.h header
408ac9e9be Manual roll Dawn from 363b4f1a0822 to b9b4a37041de (10 revisions)
3d20b817b2 Update test image to correct mistake
fec013f7b4 Roll vulkan-deps from 7625cc8362fc to a34400c4f6a0 (2 revisions)
0d91b1b40f Roll vulkan-deps from d06765b40a0b to 7625cc8362fc (4 revisions)
0fe231172a Manual roll ANGLE from 54b6968808c9 to d7ea4b2ce3c0 (7 revisions)
7456376d55 [graphite] Remove unref mutex and combine ref counts into single atomic
0f2833b305 Update SkColorType docs with bit patterns
a5c42ae18d Manual roll vulkan-deps from e54e6cd3a173 to d06765b40a0b (7 revisions)
ec32f0e67c SkCrabbyAvifCodec: Support rotation and mirroring
540db2bb3e [graphite] Fill out format tables
dc1a122178 Use templating to optimize std::function usage in Graphite PaintParams
3013d6d3f0 Roll skcms from 03700ab28aad to 12776964e26d (1 revision)
c5c98fc67c Roll vulkan-deps from fbbca36c20a6 to e54e6cd3a173 (6 revisions)
e6b5642def Roll ANGLE from c1214ec2c40d to 54b6968808c9 (5 revisions)
5891052b43 Roll Skia Infra from e0dc29520f99 to 3c369985ab80 (10 revisions)
29ff0ff0b4 Roll SwiftShader from 73053a0586b0 to a87fa39f6446 (4 revisions)
1c2a24f65d Roll Dawn from 138d7d354e31 to 363b4f1a0822 (2 revisions)
e50edb2cdb Manual roll ANGLE from 40267ddca2b8 to c1214ec2c40d (8 revisions)
06849eb99a Manual roll vulkan-deps from 5eba65d8d005 to fbbca36c20a6 (8 revisions)
8b035c3a79 [graphite] Consolidate determinations of dst texture read strategies
c8aba6a056 [graphite] Add shader snippet for ClipAtlas.
bb03124471 Skip empty chunks in SkPngCompositeChunkReader
57c433af1a Manual roll Dawn from d60d8683c9a5 to 138d7d354e31 (10 revisions)
dcf659997c [graphite] Specialization for sRGB color xforms
58ccd4f768 [Graphite] Add SkSurface::makeTemporaryImage API.
5061262298 Roll vulkan-deps from b15753b47ae5 to 5eba65d8d005 (4 revisions)
7cd26c8c4d [dawn][headers] Removes explicit MapAsync::Unknown.
3b431953b9 Add new gn location to .gitignore
b34b6f115f [Graphite] Stop letting label back to Scratch in ResourceCache.
eb9ff8e427 Use 50% load factor for `fDigestForPackedGlyphID` table
9d51bc2e06 Roll ANGLE from 3734b432018d to 40267ddca2b8 (15 revisions)
a3523a7b60 Roll Skia Infra from 11f410b4c438 to e0dc29520f99 (4 revisions)
b808498a80 Roll Dawn from 93bf2eaa5aa6 to d60d8683c9a5 (16 revisions)
ba77260fbc Roll vulkan-deps from 64530d41c091 to b15753b47ae5 (6 revisions)
c3b763fb28 Revert "Use SkCrabbyAvifCodec for HEIC and AVIF on Android"
d2db696754 [bzl] Fix //dm build
76ee21c63e [graphite] Fix up RasterPathUtils for clipping.
4ff2e3a8fe [rust png] Parse orientation from the `eXIf` chunk.
73a5aec9ad Add some documentation to the mesh_updates GM
6532b64a9e Update Skia SkCanvas docs to correct SkSurfaces methods
6f40e3c2aa Manual roll Dawn from fe1a3b8c65ed to 93bf2eaa5aa6 (8 revisions)
953cc77ad1 Manual roll SwiftShader from b99f866d5a5c to 73053a0586b0 (1 revision)
46407f56ff [infra] Add jobs for new MacMinis
c815a9b265 Manual Dawn roll
7a3eedf917 Fetch gn to third_party/gn
b52b705646 Add missing include in SkDrawTypes
03ff690028 [Fontations-backend] Roll Fontations, Skrifa to 0.26.5
8b20be96b4 Roll vulkan-deps from cbf4ac2af7a5 to 64530d41c091 (7 revisions)
96bbf45d79 Roll ANGLE from c7cfbbc90295 to 3734b432018d (7 revisions)
836fdfc670 Roll Skia Infra from 04b04dcc4949 to 11f410b4c438 (8 revisions)
22640ab9e3 Introduce SkDrawCoverage instead of error-prone boolean
7c76a89593 Mark two SkDevice subclasses as final
46ecd1d904 [graphite] Rename Resource timestamp to last use token
8a0d69cd06 [graphite] Remove Mac10.15.1 Graphite_Dawn_Metal jobs
c5dd82ebdb Roll vulkan-deps from 8d5a282ee9ba to cbf4ac2af7a5 (2 revisions)
fdfd5e5174 [graphite] Organize members of ResourceCache
5a6323775b [Fontations] Protect path retrieval arrays with lock
464a376e27 Roll vulkan-deps from a9e496961117 to 8d5a282ee9ba (6 revisions)
40c262ff88 [Fontations] Synthethize cap height
3050c2ce16 [graphite] Add static IDs for RenderSteps
e13090e31f Roll ANGLE from 0dabfe5a6ac5 to c7cfbbc90295 (9 revisions)
06677fe6a7 Roll Skia Infra from e0480d491b7b to 04b04dcc4949 (2 revisions)
60441f8c43 Roll Dawn from 50824d5923fa to c346956b0ef9 (8 revisions)
57adbff41b Roll vulkan-deps from 55e70e1178e9 to a9e496961117 (11 revisions)
7f19d763e0 Roll vulkan-deps from 127cb2fa829b to 55e70e1178e9 (1 revision)
b9b67f9b36 Roll ANGLE from 29731140b15f to 0dabfe5a6ac5 (25 revisions)
dac808134f Roll Skia Infra from 28df2c051126 to e0480d491b7b (8 revisions)
d0ebb49951 Roll Dawn from 0c8f10626e6c to 50824d5923fa (3 revisions)
ffd7fec8b6 [bzl] Fix G3 rolls
dca0ac967c Manual roll SwiftShader from f6ad529e86e4 to b99f866d5a5c (2 revisions)
cbca313620 Manual roll Dawn from 9185a4f4c58b to 0c8f10626e6c (15 revisions)
a9e8d004f0 Roll vulkan-deps from f840db3658c0 to 127cb2fa829b (14 revisions)
73341093dd Register FreeType typefaces in debugger
57abed1ab7 [bzl] Enable layering_check for Mac frameworks
5d55822e00 [graphite] Organize members in Resource.h
377ace7260 [bzl] Fix Viewer builds on Mac
d4b56ba603 Apply misc cleanups to mask filter and resource cache
c2b103c89e Fix and annotate octopus generator fiddles
eb680e6843 Roll vulkan-deps from 7e22bcde70ab to f840db3658c0 (3 revisions)
05dfc6a47c Roll Skia Infra from 0f74c98b09c6 to 28df2c051126 (6 revisions)
af80f4bb33 Roll Dawn from 617d042d4bb2 to 9185a4f4c58b (11 revisions)
1c0c5e9016 [bzl] Bump arm64 Mac toolchain to 17.0.6
d03e18f549 Roll vulkan-deps from c4ff2df7f2af to 7e22bcde70ab (4 revisions)
ea5d3fe55b [graphite] Use true scratch resources in ScratchResourceManager
367c280216 Manual roll ANGLE from de5286fa682b to 29731140b15f (2 revisions)
0a0fb30708 Deprecate (and reimpl) TightBounds
40108bcf85 Roll vulkan-deps from bd8887349203 to c4ff2df7f2af (1 revision)
05a21cbb87 [skottie] Support arbitrary displacement map sources
120379fefa Roll Skia Infra from 982a528a4385 to 0f74c98b09c6 (1 revision)
7dcc6fd6a6 Roll vulkan-deps from 3f678cea6621 to bd8887349203 (11 revisions)
d51aa116b3 Roll Skia Infra from 2f565c11a12c to 982a528a4385 (2 revisions)
efa5810875 Roll Skia Infra from 0529441d608f to 2f565c11a12c (8 revisions)
3978daa523 Roll Dawn from b7143cc4f3bf to 617d042d4bb2 (17 revisions)
488804de2c [graphite] Specialization for un/premul color xforms
2d01763761 [graphite] Add caching functionality to ClipAtlasManager
871e99c792 SkCrabbyAvifCodec: Support subsets
fb6df6a5fb Stop int overflow in TiledTextureUtils.cpp
19e7b61e39 Manual roll ANGLE from 0a207b65f60f to de5286fa682b (20 revisions)
06eb91403a [ganesh] Remove deprecated AutoCheckFlush interface.
638c27f067 [graphite] Consolidate path atlas translations into a single place.
c23e5c61e5 Roll vulkan-deps from bc80d88f6102 to 3f678cea6621 (1 revision)
8d366b3107 Break up skparagraph GNI lists
4076192f74 Extend the lifetime of SkPngCompositeChunkReader
1769d784c6 Use SkCrabbyAvifCodec for HEIC and AVIF on Android
78b812cc74 [graphite] Add Scratch shareable concept to ResourceCache
780e06a39d [bazel] Use xcrun to locate the macOS SDK
a6e93c6fc2 Roll Skia Infra from 9e9b5c63a6f8 to 0529441d608f (6 revisions)
e3dd30787e Roll Dawn from bfcb830e04ff to b7143cc4f3bf (16 revisions)
f71e1ff094 Roll vulkan-deps from d47ae2be2781 to bc80d88f6102 (10 revisions)
839e579f26 Roll skottie-base from 659bf938549f to 18c5fe2272c0
427dc976d0 Add more suppressions for ThreadedPipelineCompilePurgingTest
8d10158ffd Give RecreateSKPs 8h
123e9e7550 Roll SwiftShader from e76961fac58c to f6ad529e86e4 (4 revisions)
8b8ff96b13 Update CMake
e7b8d07885 Roll vulkan-deps from 5f7e87250bc4 to d47ae2be2781 (2 revisions)
b0427017da Roll Skia Infra from 90fa48bf59cb to 9e9b5c63a6f8 (5 revisions)
754b2cf745 Roll Dawn from 817f471a8e17 to bfcb830e04ff (26 revisions)
5b904e1b13 Roll vulkan-deps from a0c6bbad67c9 to 5f7e87250bc4 (5 revisions)
63a37d183a [rust png] Encode `pixmap.colorSpace()` as `iCCP` chunk.
c24b41eebc [graphite] Suppress threaded Precompile tests on Dawn/GLES and Dawn/Metal
3ff3711046 [bmp] Fix swizzler initialization
c6700809d2 [graphite] PrimitiveColor composes colorspace transform. Support skipColorXform for drawVertices
22f9ea65f4 [graphite] Add Pipeline cache purging
539c311a3a Roll ANGLE from f64fef257ef0 to 0a207b65f60f (14 revisions)
3b0746ea21 Roll vulkan-deps from 8540477b5d95 to a0c6bbad67c9 (1 revision)
528aa95879 Roll Skia Infra from c67b4fc704ca to 90fa48bf59cb (3 revisions)
6ddc7ba220 Roll Dawn from 8c8e27355e12 to 817f471a8e17 (13 revisions)
3531a78f28 Roll vulkan-deps from 199e36d15635 to 8540477b5d95 (16 revisions)
1398cbd6b7 Roll vulkan-deps from e264c58f9b0f to 199e36d15635 (3 revisions)
dba7f46122 Automatically scale animated webp post-decode in Android
5f2e30e0c8 Render fonts right that require hinting, as detected by Skrifa
95ed528a21 Roll ANGLE from 71980a30e08c to f64fef257ef0 (9 revisions)
66291bd9b5 Roll Skia Infra from 3c89288303c9 to c67b4fc704ca (6 revisions)
fec7d8df29 Roll Dawn from ced2fcb5d664 to 8c8e27355e12 (12 revisions)
b660e30f0a Roll vulkan-deps from 1fa2eb257bf0 to e264c58f9b0f (4 revisions)
cf47a26b74 Combine SkColorShader and SkColor4Shader
04a6578d2b [graphite] Track whether or not the Resource is reusable on the Resource
9658353bf2 Loosen G3 unsupported API Presubmit check
1d532cb013 Allow more slop in TypefaceMacVariation test
8a79cd890f SkCrabbyAvifCodec: Return correct image type
05743a92fd Revert "[ganesh] Unit test for flush of color writing for advanced blend modes"
9b556469c2 Revert "Update suppression for DMSAA_blendmodes tests"
65b12d1c82 Update suppression for DMSAA_blendmodes tests
04f0c6ef20 Add presubmit rule to check for unuspported G3 APIs
36feb25c12 [ganesh] Unit test for flush of color writing for advanced blend modes
f43d161439 [Fontations-backend] Roll Fontations, Skrifa to 0.26.3
6223c052f3 Roll vulkan-deps from 254f6951b8dc to 1fa2eb257bf0 (4 revisions)
083179bb80 Roll ANGLE from a51899d8733d to 71980a30e08c (6 revisions)
623fda5ab1 Roll Skia Infra from 2a7b1c4b1bcd to 3c89288303c9 (6 revisions)
918ba2965a Roll Dawn from 2d7fbc4b1f1b to ced2fcb5d664 (9 revisions)
c1886c4e47 Reland "[rust png] Report null color profile if only "neutral" gamma is encoded."
b44aa9b66f [graphite] Move Shareable state off key over to Resource
3ac73808d1 [ganesh] Fix the stack depth counter in GrTriangulator.
66fe3accc7 Reland "[graphite] Consolidate ycbcr sampler handling in Dawn backend"
b95665caaf Revert "Replace SkSharedMutex with std::shared_mutex"
e17eb688f4 Roll vulkan-deps from 5f05f0592633 to 254f6951b8dc (1 revision)
42a99357fb [Fontations-backend] Roll Fontations, Skrifa to 0.26.2
f8ba688521 Reduce attempts in SkOpSegment::spansNearby
40d438d766 Roll ANGLE from 465299fba5da to a51899d8733d (12 revisions)
bf74acc0a1 Roll Skia Infra from e0f9392b4eda to 2a7b1c4b1bcd (7 revisions)
fc8d77f726 Roll Dawn from 3183110b30f8 to 2d7fbc4b1f1b (10 revisions)
8bbd583477 Roll vulkan-deps from a652576bff1b to 5f05f0592633 (12 revisions)
c09407add3 Make defensive copies of SkPath less frequently when drawing them
e944b50469 Revert "[rust png] Report null color profile if only "neutral" gamma is encoded."
8aed02367a [skottie] Support nonsequential track mattes
7cf08825d7 Revert "[graphite] Consolidate ycbcr sampler handling in Dawn backend"
0d70ea7bf0 [rust png] Report null color profile if only "neutral" gamma is encoded.
e995e9f190 [graphite] ResourceProvider defines Sampler resource key
3d1ecf63e5 Reland "Limit stack calling depth from GrTriangulator::mergeCollinearEdges()."
42e38a8610 [graphite] Consolidate ycbcr sampler handling in Dawn backend
8d5fabf4f8 [graphite] Consolidate vulkan ycbcr conversion handling
9b3395edd0 Roll vulkan-deps from e2e618fdd93a to a652576bff1b (8 revisions)
52f7ba884f Merge 9 release notes into RELEASE_NOTES.md
50e6fe5901 Update Skia milestone to 134
e71ab89812 Roll ANGLE from d65751b40f7e to 465299fba5da (13 revisions)
f820e6f330 Roll Skia Infra from 783200f5fd9e to e0f9392b4eda (3 revisions)
c99770530d Roll Dawn from 3f908f3907e5 to 3183110b30f8 (10 revisions)
5310fc642c Roll vulkan-deps from 0cf38d98a8ba to e2e618fdd93a (7 revisions)
9799dcab37 [webp] Use the same RP blender for all lines
279fde045c GraphiteDawn: Don't enable SSBO in Compat Mode.
77f4941092 Set underlying type of SkFontStyle::Slant
4a7628f0c8 In play_skottie.cpp example check that surface was allocated
84d9fd6e3b SkFontMgr_FCI refactoring
20164f3db7 [Graphite] Implement Vulkan query to know lazy allocation size.
3ee39f3584 Allow MSKP benches to throttle frames on GPU.
472faa80de Roll ANGLE from b225419fd227 to d65751b40f7e (2 revisions)
1c76d8b280 Roll vulkan-deps from 85de9aa78bc8 to 0cf38d98a8ba (2 revisions)
58c6de0423 Roll Skia Infra from 6fd1ead44af5 to 783200f5fd9e (3 revisions)
71e698aedf Roll Dawn from 28858cfac0b9 to 3f908f3907e5 (1 revision)
cb18e028c4 Roll vulkan-deps from d2e10146ec6f to 85de9aa78bc8 (1 revision)
b4613bade5 Roll vulkan-deps from a09312dc04d3 to d2e10146ec6f (1 revision)
babfedf497 Manual roll Dawn from 7f24ec953f26 to 28858cfac0b9 (7 revisions)
975788ea90 [graphite] Fix clients that depend on MtlGraphiteTypes.h
34c784a292 Remove approximateFilteredBounds from SkMaskFilter
5857fc2fe4 Use skia_rules.bzl instead of macros.bzl where possible
ac9c72dcf0 [graphite] Metal backend type utility functions callable from C++
fbc37088ac Ganesh uses default bazel obj copts, Graphite explicit about no ARC.
df3f49098f Expose skottie for external Bazel clients.
b8e83110f0 Roll vulkan-deps from 57295eb4b939 to a09312dc04d3 (1 revision)
44ef31b19d Roll ANGLE from 9d9e3efce077 to b225419fd227 (2 revisions)
fffbc97d9d Roll Skia Infra from ec5e5e9d35c8 to 6fd1ead44af5 (1 revision)
efbf755499 Roll Dawn from 5b33b8d8385d to 7f24ec953f26 (4 revisions)
8cfeb42c62 Roll vulkan-deps from b7e7b5c365d1 to 57295eb4b939 (3 revisions)
3123598acb [pdf] Allow simple /StructElem /K
24365abf85 [rust png] Performance: Cache and reuse `fColorTable` if possible.
d2e8363d13 [rust png] Avoid doing alpha premultiplication twice.
fe8bac8f22 Remove unneeded resource refs on Vulkan CommandBuffer.
8eebb997a5 Update viewer builds to work on newest android devices.
65d7bff9d1 [rust png] Use `u32` => `f32` conversion that matches Blink.
68d4407438 Roll vulkan-deps from f45ce3e3fa15 to b7e7b5c365d1 (5 revisions)
8670a9edba Roll ANGLE from 6015e7305d15 to 9d9e3efce077 (3 revisions)
095e1a7961 Roll Skia Infra from d359a715b7ed to ec5e5e9d35c8 (1 revision)
bb06d0aa57 Roll Dawn from 28b186286761 to 5b33b8d8385d (2 revisions)
693030a1df Roll vulkan-deps from 04a50a3dd7b9 to f45ce3e3fa15 (1 revision)
e99b226c3d Roll ANGLE from 39882b2b735f to 6015e7305d15 (2 revisions)
31d19df840 Roll Skia Infra from d10e2fbb052e to d359a715b7ed (2 revisions)
d59c150674 Roll Dawn from e5701af2e856 to 28b186286761 (1 revision)
e045d7ef5e Roll vulkan-deps from bfa23916c97e to 04a50a3dd7b9 (2 revisions)
09f8ab0c9b Roll vulkan-deps from 93cfcbf427b3 to bfa23916c97e (2 revisions)
5e8c3cb793 Roll ANGLE from 017a9382befe to 39882b2b735f (1 revision)
89f1df000d Roll Skia Infra from 9dab19606d40 to d10e2fbb052e (4 revisions)
d1c208fb5e Roll Dawn from dc4935b883dd to e5701af2e856 (2 revisions)
c260865988 Manual roll ANGLE from c0ee7b203bad to 017a9382befe (40 revisions)
efe8bac7d1 Roll vulkan-deps from 4cc576cd0e3d to 93cfcbf427b3 (1 revision)
787f004498 Roll Dawn from db55f5957411 to dc4935b883dd (5 revisions)
fddefe73c5 Roll Skia Infra from 1ed863c4385b to 9dab19606d40 (6 revisions)
0ec5145fca Roll SwiftShader from 118f577b4188 to e76961fac58c (1 revision)
b5839e6326 Roll vulkan-deps from 6455a044bee9 to 4cc576cd0e3d (1 revision)
e107cb001d Roll vulkan-deps from 2b029290ba80 to 6455a044bee9 (4 revisions)
33a9a4ffae Roll vulkan-deps from b7874f7020d2 to 2b029290ba80 (1 revision)
44b9707fc1 Roll vulkan-deps from 2ceb995d091f to b7874f7020d2 (1 revision)
7a4ef0b9a2 Roll Skia Infra from b95a7b5a23b2 to 1ed863c4385b (5 revisions)
db37ded7ce Roll Dawn from 76129f64155f to db55f5957411 (2 revisions)
22127449d7 Roll Skia Infra from 56a20669db21 to b95a7b5a23b2 (1 revision)
9ce064f60d Roll Dawn from 45926030e9e1 to 76129f64155f (2 revisions)
ccefbb7a59 Roll Skia Infra from b0b99f248aab to 56a20669db21 (1 revision)
daecd808dc WebGPU: Use VertexStepMode Undefined instead of VertexBufferNotUsed
7b16ec6d9c Roll Skia Infra from 2ad74dabbee5 to b0b99f248aab (6 revisions)
c0c3d52f29 Roll Dawn from 23bb5c065341 to 45926030e9e1 (3 revisions)
bdbc037fd6 Roll vulkan-deps from d9e522973bf6 to 2ceb995d091f (2 revisions)
1c9c95eb6f Keep last resort glyphs and positions together
e0114876f8 Only output last valid scan of progressive JPEG decode
79f1fd9a9d [infra] Add jobs for new MacBookPro15,3 machines
0440559439 Roll vulkan-deps from dd42fc42309c to d9e522973bf6 (3 revisions)
46df6fed41 Roll Skia Infra from 0abec67a371b to 2ad74dabbee5 (7 revisions)
f0bf1fd96d Roll Dawn from ef00af577667 to 23bb5c065341 (1 revision)
1949fee840 Roll vulkan-deps from 0f01167cd48b to dd42fc42309c (1 revision)
d5f0cf5a7c Roll vulkan-deps from f6642af20ac1 to 0f01167cd48b (2 revisions)
cf67bf31e0 Manual roll Dawn from bd302da5fe83 to ef00af577667 (10 revisions)
47957be04e Revert "Limit stack calling depth from GrTriangulator::mergeCollinearEdges()."
eb33dd41db Roll vulkan-deps from b51f6c0aa71b to f6642af20ac1 (11 revisions)
a38c4ae028 Limit stack calling depth from GrTriangulator::mergeCollinearEdges().
005f65a914 [graphite] Disable use of CommandBufferTimestamps for dawn-mac
40959c4dbc [Graphite] Add support in the ResourceCache for change a Resources size.
a0c37ec4ca [graphite] Adjust raster path key to include clip origin.
a15d02f775 Update the test 'FontMgr_Iter'
99dd91b4fe Remove CoreText 14 and 15 work arounds
2521ff091d [infra] Move to Android devices in the new lab
70d482d240 Roll vulkan-deps from 33f530ef3285 to b51f6c0aa71b (3 revisions)
c88fc17d1d Roll Dawn from c3cd3b4a6d40 to bd302da5fe83 (7 revisions)
38461b444d Roll Skia Infra from fb8dd36529f6 to 0abec67a371b (9 revisions)
e35527e40d Roll vulkan-deps from e616c9b1ad31 to 33f530ef3285 (4 revisions)
efebb9d95e Make size pre-allocated for SkCanvas::save() controllable in SkUserConfig
96b37e2a3e [viewer] Check for valid WindowContext in Window::supportsGpuTimer
8f33adb1ac Adjust slant value for 'CTFontDescriptor' to improve font matching
ee9f7aec12 fiddle_main.h to not include skia.h
d746d5faaa Add SK_NO_SANITIZE_CFI that has effect when building with Clang
8c518c6d3a [graphite][Dawn] Add missing attribute from YCbCrVkDescriptor comparison
60f1c30d07 Clarify size of SkCanvas's stack allocation
69ebfded97 [infra] Replace RadeonVega6 jobs with RadeonVega8
e5296cebaa [graphite] Add Vulkan push constant support
1d9cace0d9 Manual roll vulkan-deps from 491933b72060 to e616c9b1ad31 (7 revisions)
e3b2835099 Roll vulkan-deps from ae8417f428a1 to 491933b72060 (5 revisions)
20e3d71981 Roll Skia Infra from 7544f7c1ccbb to fb8dd36529f6 (10 revisions)
44e33b331d Roll Dawn from cc2acb18f2a8 to c3cd3b4a6d40 (13 revisions)
10a65a187b Roll SwiftShader from 23b93c7cfe27 to 118f577b4188 (1 revision)
b2f382eaae [rust png] More granular gni list: `skia_encode_libpng_srcs`.
cdd807647d [rust png] Update `cxx` from 1.0.131 to 1.0.135.
85baab670f Remove skiaserve and libmicrohttpd
a6ea45cb42 [graphite] Establish exclusive descriptor set for Vulkan input attachments
248b5242d0 Support correct flushing for Graphite in MSKPBench
4a9d5e24dd Roll skottie-base from 32e4afed6d80 to 659bf938549f
fc205a7656 [viewer] Fix captured reference to stack variable
8f2a29de51 Fix TODO in SkRRect about dense packing
492a9e761c Roll vulkan-deps from dbec774c7f68 to ae8417f428a1 (2 revisions)

Canonical link: https://commits.webkit.org/289878@main
